pkg/ccl/crosscluster/logical/logical_replication_writer_processor.go

```diff
@@ -991,6 +998,9 @@ func (t *txnBatch) HandleBatch(
 	} else {
 		err = t.db.Txn(ctx, func(ctx context.Context, txn isql.Txn) error {
 			for _, kv := range batch {
+				if t.spec.Discard == jobspb.LogicalReplicationDetails_DiscardAllDeletes && len(batch[0].KeyValue.Value.RawBytes) == 0 {
+					continue
+				}
```

Comment:
I think `batch[0]` should probably be `kv` here.


```suggestion
				if t.shouldDiscardDeletes() && !kv.Value.IsPresent() {
					continue
				}
```